save new config filename/location

RetroArch 1.3.1 / 7d8ad3e

Launching with

```
/opt/retropie/emulators/retroarch/bin/retroarch -L /opt/retropie/libretrocores/lr-mame2003/mame2003_libretro.so --config /opt/retropie/configs/all/retroarch.cfg ~/RetroPie/roms/mame-libretro/rom.zip
```

retroarch.cfg - http://pastebin.com/rQYAkBX5

on launch it flashes up the error "failed to save config /opt/retropie/configs/all/retroarch.cfg (why is it trying to save the config on launch? or is this just a false error). The file is writeable.

when entering RGUI and choosing "Save New Config" it tried to save a file called ".cfg" in the current folder (rather than the location of the config file which would be preferable). if doing it again it chooses the name -1.cfg, then -2.cfg

I believe this used to work (it would choose a more sensible filename) and used to use the config folder location.

hmm i corrected the permissions problem I thought I had, and made sure I can write the file and it's still trying to write on start and still failing. odd

```
RetroArch [INFO] :: [CPUID]: Features:
RetroArch [ERROR] :: Failed saving config to "/opt/retropie/configs/all/retroarch.cfg".
```

I will debug further then.

--- a/command_event.c
+++ b/command_event.c
@@ -702,7 +702,7 @@ static bool event_save_core_config(void)
          else
             strlcpy(tmp, ".cfg", sizeof(tmp));
 
-         snprintf(config_path, sizeof(config_path), "%s%s", config_path, tmp);
+         strlcat(config_path, tmp, sizeof(config_path));
          if (!path_file_exists(config_path))
          {
             found_path = true;

